mailbox/elasticsearch/src/main/java/org/apache/james/mailbox/elasticsearch/MailboxMappingFactory.java

```diff
@@ -83,13 +84,42 @@ public static XContentBuilder getMappingContent() {
                 .startObject()
 
                     .startObject(MailboxElasticsearchConstants.MESSAGE_TYPE.getValue())
-                        .startObject(PROPERTIES)
+                        .field(NodeMappingFactory.DYNAMIC, false)
 
+                        .startObject(JsonMessageConstants.PROPERTIES)
                             .startObject(MESSAGE_ID)
                                 .field(TYPE, STRING)
                                 .field(INDEX, NOT_ANALYZED)
                             .endObject()
 
+                            .startObject(JsonMessageConstants.HEADERS)
+                                .field(NodeMappingFactory.TYPE, NodeMappingFactory.NESTED)
+                                .startObject(NodeMappingFactory.PROPERTIES)
+                                    .startObject(JsonMessageConstants.Header.KEY)
+                                        .field(NodeMappingFactory.TYPE, NodeMappingFactory.STRING)
+                                        .field(NodeMappingFactory.INDEX, NodeMappingFactory.NOT_ANALYZED)
+                                    .endObject()
+                                    .startObject(JsonMessageConstants.Header.VALUE)
+                                        .field(NodeMappingFactory.TYPE, NodeMappingFactory.STRING)
+                                        .field(NodeMappingFactory.INDEX, NodeMappingFactory.NOT_ANALYZED)
+                                    .endObject()
+                                .endObject()
+                            .endObject()
+
+                            .startObject(JsonMessageConstants.ATTACHMENTS)
```

Comment:
Ok I did not think about attachments when analyzing the mapping. But we had the same inssu than header nothing was specified, and attachment also have a header field with possibly an infinity of different header key.
In this PR, I did the minimum concerning the specification of ATTACHMENTS to pass unit test. 
Should I do more ?